intro/projects-data/projects-data-nb.ipynb

```diff
@@ -0,0 +1,248 @@
+   "id": "d6a05b7c-4110-4132-8a2c-5433b3fb88d3",
+   "metadata": {},
+   "source": [
+    "5. Use the connection to retrieve the list of available buckets."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "411296b4-cd8d-4e65-a352-fab9c478e418",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "response = s3.list_buckets()\n",
+    "response[\"Buckets\"]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d2223269-a61e-4f91-addd-9af197b5a7f8",
+   "metadata": {},
+   "source": [
+    "6. Upload a file to the bucket via the connection."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b25303ea-cd23-4958-b08a-c5a52beda700",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create a file-like object containing bytes that represent the \"hello world\" string\n",
+    "file_obj = io.BytesIO(b\"hello world\")\n",
+    "\n",
+    "# upload the file-like object to the S3 bucket specified in the data connection\n",
+    "# the name of the \"file\" in S3 is \"hello.txt\"\n",
+    "s3.upload_fileobj(file_obj, bucket_name, Key=\"hello.txt\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7b407ad7-14b2-4149-9220-a7188fd3a05c",
+   "metadata": {},
+   "source": [
+    "6. List the contents of the bucket specified in the data connection."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1c0a4879-ed47-4132-8abb-7fcd42cb0af7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# retrieve the metadata of contents within the bucket\n",
+    "objects = s3.list_objects_v2(Bucket=bucket_name)\n",
+    "\n",
+    "# output the name of each object within the bucket\n",
+    "for obj in objects[\"Contents\"]:\n",
+    "    print(obj[\"Key\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "63f252dd-3abe-4dd9-ad09-282047b6d97a",
+   "metadata": {},
+   "source": [
+    "> NOTE: Optionally, check the corresponding S3 bucket for the new `hello.txt` object."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b9e8cd99-7644-4d02-849b-c30bd744ffaa",
+   "metadata": {},
+   "source": [
+    "7. Download the file from the S3 bucket to a new location."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0c1ea1b1-dc8e-4027-bc56-75a2ce8084d6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s3.download_file(bucket_name, \"hello.txt\", \"new_hello.txt\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c1c591fb-f880-401c-b52f-e719ae83c753",
+   "metadata": {},
+   "source": [
+    "8. The pane to the left displays a new file called `new_hello.txt`.\n",
+    "Open the file and verify that its contents are `hello world`.\n",
+    "\n",
+    "> NOTE: You might need to refresh the file browser by clicking the `Refresh the file browser` button in the file browser pane.\n",
+    "The button displays as a circular arrow."
+   ]
+  },
```

Comment:
Can we add a final markdown cell saying something like this?

> Return to the course book and finish the exercise.

I made the same suggestion in Alejandro's PR cause I forgot to go back and run the lab finish script.

https://github.com/RedHatTraining/AI26X-apps/pull/4/files#diff-873c83ef199f002867685dd407805c4428c2b39444e440e6b90b56fc982531d0R286